modules/gapi/src/backends/cpu/gcpubackend.cpp

```diff
@@ -78,11 +78,25 @@ cv::gimpl::GCPUExecutable::GCPUExecutable(const ade::Graph &g,
 {
     // Convert list of operations (which is topologically sorted already)
     // into an execution script.
+    GConstGCPUModel gcm(m_g);
     for (auto &nh : nodes)
     {
         switch (m_gm.metadata(nh).get<NodeType>().t)
         {
-        case NodeType::OP: m_script.push_back({nh, GModel::collectOutputMeta(m_gm, nh)}); break;
+        case NodeType::OP:
+        {
+            m_script.push_back({nh, GModel::collectOutputMeta(m_gm, nh)});
+
+            GCPUKernel k = gcm.metadata(nh).get<Unit>().k;
+            if (k.m_isStateful)
+            {
+                // Initial state setup
+                m_nodesToStates[nh] = GArg { };
+
+                setupKernelState(nh, m_nodesToStates[nh]);
```

Comment:
Two map look-ups instead of one. Please rework.

Reply:
Fixed, thanks!